### PR TITLE
ruby: add v3.3.4

### PR DESCRIPTION
**Pull request by TApplencourt**
Add Ruby 3.3.4 (https://www.ruby-lang.org/en/downloads/)

### CODE REVIEW

**Review by bernhardkaindl — DISMISSED** [review]

**Review by Kerilk — APPROVED** [review]

File: var/spack/repos/builtin/packages/ruby/package.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Ruby(AutotoolsPackage, NMakePackage):
     tags = ["windows"]
 
     license("Ruby AND BSD-2-Clause AND MIT", checked_by="tgamblin")
-
+    version("3.3.4", sha256="fe6a30f97d54e029768f2ddf4923699c416cdbc3a6e96db3e2d5716c7db96a34")
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/ruby/package.py]: @TApplencourt following the common package style, we should keep the line break between the 
`license()` and `version()` directives.

**Review by alecbcs — CHANGES_REQUESTED** [review]
One quick request. Otherwise this PR looks good to me.

File: var/spack/repos/builtin/packages/ruby/package.py (hunk 1 above)
Comment (TApplencourt) [reply var/spack/repos/builtin/packages/ruby/package.py]: I  think the line break broke `black`, hence my change. (see d4be275 and https://github.com/spack/spack/actions/runs/10014130150/job/27683195517) or I missread the diff?

Comment (alecbcs) [reply var/spack/repos/builtin/packages/ruby/package.py]: `black` removed trailing whitespace on the empty line. An empty line without spaces on it should be fine.

Comment (TApplencourt) [reply var/spack/repos/builtin/packages/ruby/package.py]: Oh! Thanks!

**Review by alecbcs — APPROVED** [review]
Confirmed SHA. Looks good to me. Thanks @TApplencourt

**Pull request merged**